Reduce sandbox telemetry from WebContent process on iOS

pvollan opened the pull request:
#### 568ff7623f7a79a2fefbf96f9c840a3f7ea4e096
<pre>
Reduce sandbox telemetry from WebContent process on iOS
<a href="https://bugs.webkit.org/show_bug.cgi?id=272927">https://bugs.webkit.org/show_bug.cgi?id=272927</a>
<a href="https://rdar.apple.com/126656229">rdar://126656229</a>

Reviewed by Brent Fulgham.

Allow 2 syscalls to be used, but only during launch of the WebContent process.
Silence reports from 2 IOKit related syscalls, since all IOKit is blocked in
the WebContent process.

* Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.WebContent.sb.in:

Canonical link: <a href="https://commits.webkit.org/277760@main">https://commits.webkit.org/277760@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1bba99940785d5d298edfff12f94cb5ab86ecd38

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/48441 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/27653 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/51400 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/51129 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/44506 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/33589 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/25175 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/39598 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/49023 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/25332 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/41814 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/20719 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/22816 "Passed tests") | | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/6498 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/44757 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/43465 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/53034 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/23488 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/19817 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/46911 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/24753 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/42010 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/45827 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/25558 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/6907 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/24476 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

brentfulgham commented on Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.WebContent.sb.in:
```diff
@@ -1091,7 +1091,7 @@
         SYS_write))
 
-(deny syscall-unix (with telemetry))
+(deny syscall-unix)
```
Should we just silence the `SYS_Persona` call? Or do we see many others we don't want to learn about?

brentfulgham commented on Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.WebContent.sb.in:
```diff
@@ -1392,7 +1392,7 @@
 (allow mach-kernel-endpoint
     (apply-message-filter
-        (deny mach-message-send (with telemetry))
+        (deny mach-message-send)
```
Likewise, should we just silence the one item we know is noisy but perhaps leave others?

pvollan replied:
That is a very good point. I have adjusted the patch accordingly.

Thanks for reviewing!

brentfulgham reviewed (approved):
This change is clearly safe, and will resole the telemetry issue. But I wonder if it would be more useful to simply silence the items that we know are noisy, so we can surface cases we need to know about?

brentfulgham reviewed (approved):
Oh, this is much nicer. Looks great! r=me

pvollan committed:
Reduce sandbox telemetry from WebContent process on iOS
https://bugs.webkit.org/show_bug.cgi?id=272927
rdar://126656229

Reviewed by Brent Fulgham.

Allow 2 syscalls to be used, but only during launch of the WebContent process.
Silence reports from 2 IOKit related syscalls, since all IOKit is blocked in
the WebContent process.

* Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.WebContent.sb.in:

Canonical link: https://commits.webkit.org/277760@main
